x-pack/plugins/stack_alerts/server/rule_types/geo_containment/executor.ts

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { RuleExecutorOptions } from '../../types';
+import {
+  canSkipBoundariesFetch,
+  executeEsQuery,
+  getEntitiesAndGenerateAlerts,
+  getRecoveredAlertContext,
+  getShapeFilters,
+  transformResults,
+} from './lib';
+import type {
+  GeoContainmentRuleParams,
+  GeoContainmentAlertInstanceState,
+  GeoContainmentAlertInstanceContext,
+  GeoContainmentRuleState,
+} from './types';
+
+import { ActionGroupId, GEO_CONTAINMENT_ID } from './constants';
+
+export async function executor({
+  previousStartedAt,
+  startedAt: windowEnd,
+  services,
+  params,
+  rule,
+  state,
+  logger,
+}: RuleExecutorOptions<
+  GeoContainmentRuleParams,
+  GeoContainmentRuleState,
+  GeoContainmentAlertInstanceState,
+  GeoContainmentAlertInstanceContext,
+  typeof ActionGroupId
+>): Promise<{ state: GeoContainmentRuleState }> {
+  const boundariesRequestMeta = {
+    geoField: params.geoField,
+    boundaryIndexTitle: params.boundaryIndexTitle,
+    boundaryGeoField: params.boundaryGeoField,
+    boundaryNameField: params.boundaryNameField,
+    boundaryIndexQuery: params.boundaryIndexQuery,
+  };
+  const { shapesFilters, shapesIdsNamesMap } =
+    state.shapesFilters &&
+    canSkipBoundariesFetch(boundariesRequestMeta, state.boundariesRequestMeta)
```

Comment:
Snippet from the docs (my emphasis italicized)

> Boundaries index or data view: An index containing geo_shape data, such as boundary data and bounding box data. This data is presumed to be static (not updating). Shape data matching the query is harvested once when the rule is created _and anytime after when the rule is re-enabled after disablement_. 

This doesn't seem to update the boundaries in the state when I disable and re-enable the rule. It looks like this is also an issue in `main`. If you feel it is out of scope for this PR then let's open an issue to track this.

Reply:
lets open an issue. I will need to research how to tell if rule is re-enabled and I think these changes are worth merging until we have time to research this. Thanks for raising this concern.

Reply:
created https://github.com/elastic/kibana/issues/157933 to track